### ISSUE TITLE
KeyError: ('___BEGIN__', '___BEGIN__')

### ISSUE THREAD

**Issue by icyphox** [jsvine/markovify#92]
I'm using a simple `text_model.make_short_sentence(140)`. Nothing new here.

I can't seem to figure this out, it was working just fine. Here's the traceback:
```
Traceback (most recent call last):
  File "bot.py", line 51, in <module>
    main()
  File "bot.py", line 34, in main
    markov_text = marko.gen_markov(f=filepath)
  File "/home/icyphox/code/memebot/modules/marko.py", line 18, in gen_markov
    model = markovify.Text(text)
  File "/home/icyphox/code/memebot/.env/lib/python3.6/site-packages/markovify/text.py", line 37, in __init__
    self.chain = chain or Chain(self.parsed_sentences, state_size)
  File "/home/icyphox/code/memebot/.env/lib/python3.6/site-packages/markovify/chain.py", line 45, in __init__
    self.precompute_begin_state()
  File "/home/icyphox/code/memebot/.env/lib/python3.6/site-packages/markovify/chain.py", line 80, in precompute_begin_state
    choices, weights = zip(*self.model[begin_state].items())
KeyError: ('___BEGIN__', '___BEGIN__')
```

**Comment by kade-robertson** [comment]
Usually means there was some kind of symbol it doesn't support (parenthesis for example).

**Comment by jsvine** [comment]
Thanks for pointing to #84, @kade-robertson! That does seem to be a likely explanation. 

The issue, however, is typically less that the corpus *contains* an unsupported character, and more that *all* sentences contain such a character. But it's a bit difficult to diagnose definitively without knowing more about the corpus. Could you share the corpus you're using @icyphox, or a sample of it?

**Comment by icyphox** [comment]
I'm actually generating the corpus from a user's Twitter feed. And that's most likely it, as @kade-robertson said.

**Comment by kade-robertson** [comment]
I've made #93 is a partial workaround to the issue, you can disable the rejection regex entirely with `well_formed=False` in `Text` initialization, or provide a custom regex with `reject_reg` to substitute. This may not be ideal as you will then get unmatched parenthesis in the output. There's no one good way to solve the issue unfortunately -- You either don't use that text at all for learning purposes, or you enable unmatched parenthesis or weird punctuation.

**Comment by crypdick** [comment]
@kade-robertson I am experiencing this problem. I see that adding `well_formed=False` [was added](https://github.com/kade-robertson/markovify/blob/abf6d770cd444e134de64a7cbf40d5a16ce2726e/markovify/text.py#L19) but for some reason  I don't see that option in the main branch anymore?

**Comment by kade-robertson** [comment]
@crypdick @icyphox #93 was just recently merged in if you are still trying to use this library with your original corpus. :)

**Comment by icyphox** [comment]
Oh that’s awesome! although my Twitter bot is long dead. :D

Guess I’ll close this now.

**Comment by keli5** [comment]
I'm getting this issue when trying to use a corpus of well-formed utf-8 text, albeit full of parentheses. Whether I use well_formed true or false while building the model, I get the same error:
```
  File "C:\Users\letters\AppData\Roaming\Python\Python310\site-packages\markovify\text.py", line 65, in __init__
    self.chain = chain or Chain(self.parsed_sentences, state_size)
  File "C:\Users\letters\AppData\Roaming\Python\Python310\site-packages\markovify\chain.py", line 53, in __init__
    self.precompute_begin_state()
  File "C:\Users\letters\AppData\Roaming\Python\Python310\site-packages\markovify\chain.py", line 102, in precompute_begin_state
    choices, cumdist = compile_next(self.model[begin_state])
KeyError: ('___BEGIN__', '___BEGIN__')
```
